src/Process/Runner/MutationTestingRunner.php

```diff
@@ -103,6 +112,22 @@ public function run(iterable $mutations, string $testFrameworkExtraOptions): voi
 
                 return false;
             })
+            ->filter(function (Mutant $mutant) {
+                $mutatorName = $mutant->getMutation()->getMutatorName();
+
+                if (!array_key_exists($mutatorName, $this->ignoreSourceCodeMutatorsMap)) {
+                    return true;
+                }
+
+
+                foreach ($this->ignoreSourceCodeMutatorsMap[$mutatorName] as $sourceCodeRegex) {
+                    if (preg_match("/-(.*?){$sourceCodeRegex}/", $mutant->getDiff()) === 1) {
```

Comment:
What's the catch here?

Reply:
not really. Since we are applying reges on the diff, we want to apply it on the *originall* code, this is why we have a `-` on the begging.

in the e2e test of this PR, with the following config

```
"ignoreSourceCodeByRegex": [
    "getString"
]
```

we are ignoring such mutation:

```diff
- public function getString()
+ public function getString()
```

as you can see, this is not the beginning of the string